### ISSUE TITLE
Convert unencrypted Folders to Encrypted & vice versa

### ISSUE THREAD

**Issue by psi5asp** [syncthing/syncthing#7886]
### Include required information
With existing shared folders between devices, have the function to automatically convert the existing folder from unencrpted (Trusted) to encrypted (untrusted) when entering a password for that folder (and/or device?).

Currently with existing shares, if a device becomes untrustworthy (ie loose control over, someone else takes ownership, or some event that is outside the control of the admin), one cannot do much about it. Removing the folder and re-sharing as untrustworth still keeps the original files on the (now) untrusted device.

Also when converting a folder share from trusted to untrustworthy, it is a long manual task of removing the share, removing the (now) untrustwworthy device/folder files on the h/d (if one still has access to it), then re-adding it as 'untrusted device' and starting the whole re-sync process again (add, accept, resync). This is a nightmare when dealing with many folders and many devices.
 
If syncthing can do all the conversion of the existing content on the (now) untrusted device this would save a lot of time;
eg Device A (sending device) - Device B (Recieving Device)
1) Device A changes Device B to 'Untrsted' by adding a password to folder for the Device B
2) Device A starts its encrption proccess to start sending the encrpted files to Device B.
3) Device A prevents any further action on Device B folder (since its in the proccess of converting the conents) unless manaully requested to abort and reverse the process.
4) As device B recieves new encrypted folder content, device A is requesting Deive B to remove the orignal data once each file is successfully converted and transfered
5) Once completed, the original files on Device B are all removed, and syncing resumes as normal like an untrusted device folder.

The conversion from Untrsted to Trusted is done as above, but from Encrypted to Decrypted on the Sender (Device A) side, and one at a time file transfer-validate is done.

**Comment by AudriusButkevicius** [comment]
> Currently with existing shares, if a device becomes untrustworthy (ie loose control over, someone else takes ownership, or some event that is outside the control of the admin), one cannot do much about it. Removing the folder and re-sharing as untrusted still keeps the original files on the (now) untrusted device. This is a major security concern.

There is nothing you can do about that. For all that you know, the other side might change the folder to send only, and the fact you added a password etc, has zero meaning. 

There might be another device the same folder is shared, which doesn't have a password etc.

In reality what would happen, is you'd start getting connection errors I think, as the expected folder types would not match, and not suddenly make the folder encrypted on the other side.

I think you are looking at this through the eyes of centralised control, whereby you can force some other device to do something. Syncthing is decentralised, so if one of your devices is compromised, there is nothing you can do from other devices to prevent the damage. You say its a security concern, but it's the cost of a decentralised system.

I don't think it's viable for us to support the process of conversion while syncthing is running, and I think this would have to be done via a separate cli command, while it's not running.

There are a ton of things that could happen that would interrupt/screw up the process that might lead to you losing your data.


You have to use the CLI to do this.

Doing this from the UI, while its running, is unfeasible, because first

**Comment by acolomb** [comment]
I agree this would be first class security theater. If someone who already has all your data unencrypted (with the power to duplicate or distribute it to others) becomes no longer trustworthy, there is nothing you can do to stop them. Having your software ask their software politely do replace the data gives a false sense of security if anything.

**Comment by psi5asp** [comment]
That was only 1 possible case use. Automating the manual conversion is the main reason why this request was raised....

But like the other requests I'm sure this feature request will get rejected because of 1 persons' fixed ideas and without proper consideration.

2c

**Comment by AudriusButkevicius** [comment]
I think you are throwing out claims like that without understanding how syncthing works or the design ideas behind it.

One of the first words in the marketing pitch is "decentralised" meaning no-one can force someone else to do something.

It's not about my personal views, its about the design philosophy of the application.

The fact that you want something different doesn't mean I think it would not be useful, it probably would, I might find something like that useful myself, but that doesn't suddenly make it possible to implement in syncthing, or compatible with design philosophy of syncthing.

Lastly, in the scenario you describe, syncthing could be used as "ransomware" if compromised, and I don't think we want that.

**Comment by AudriusButkevicius** [comment]
Never the less, I agree the conversion aspect now is grim, and could/should be streamlined, hence why this is still open.

**Comment by w1v1** [comment]
There is a difference between Content/File management and Decentralised systems and they don't necessarily relate, since content management deals with ownership and control of files. In our organisation this would be a factor in using a system that can't tell the difference.

**Comment by mbomb007** [comment]
Just a thought, this feature would also need to prevent ransomware-like behavior. An untrusted device shouldn't be able to change your folders from unencrypted to encrypted.
